components/org.wso2.carbon.identity.sso.agent.saml/src/main/java/org/wso2/carbon/identity/sso/agent/saml/bean/SSOAgentConfig.java

```diff
@@ -296,6 +307,22 @@ public void initConfig(Properties properties) throws SSOAgentException {
                 skipURIs.add(skipURI);
             }
         }
+        if (!StringUtils.isBlank(properties.getProperty(SSOAgentConstants.SSOAgentConfig.INDEX_PAGE))) {
+            setIndexPage(properties.getProperty(SSOAgentConstants.SSOAgentConfig.INDEX_PAGE));
+            skipURIs.add(indexPage);
+        } else {
+            StringBuilder stringBuilder = new StringBuilder();
```

Comment:
Let's extract this to a separate method. Also let's use a URL builder rather doing direct string manipulation.

Reply:
Sure, thanks.